### PR TITLE
chore(non-null-assertions): addressed some non-null-assertions

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves [] 

### Description

Addressing some non-null-assertions - PLEASE raise your concerns 😅

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- addressed some non-null-assertions - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

**Comment by ArtsyOpenSource** [comment]
### This PR contains the following changes:

- Dev changes (addressed some non-null-assertions - gkartalis)
- undefined (changes)
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d965fd1bbea02499287f7818dc40a1fdb861c001
</p>

File: src/app/Scenes/Feature/components/FeatureFeaturedLink.tsx [hunk 1]
```diff
@@ -17,13 +17,12 @@ const FeatureFeaturedLink: React.FC<FeatureFeaturedLinkProps> = ({ featuredLink,
...
-                navigate(featuredLink.href!)
-              }
-            : undefined
-        }
+        onPress={() => {
+          if (!featuredLink.href) {
+            return
+          }
+          navigate(featuredLink.href)
+        }}
```
Comment (araujobarret) [inline src/app/Scenes/Feature/components/FeatureFeaturedLink.tsx:25]:
```suggestion
        onPress={() => {
          if (featuredLink.href) {
            navigate(featuredLink.href)
          }
        }}
```

**Review by araujobarret — DISMISSED** [review]
Lots of good ones!

Left a comment if we have more cases like that would be nice to refactor like the example.

File: src/app/Scenes/ArtistSeries/ArtistSeriesMoreSeries.tsx [hunk 2]
```diff
@@ -62,7 +62,7 @@ export const ArtistSeriesMoreSeries: React.FC<ArtistSeriesMoreSeriesProps> = ({
           <TouchableOpacity
             onPress={() => {
               trackEvent(tracks.tapViewAllArtistSeries(artist?.internalID, artist?.slug))
-              navigate(`/artist/${artist?.internalID!}/artist-series`)
+              navigate(`/artist/${artist?.internalID}/artist-series`)
```
Comment (brainbicycle) [inline src/app/Scenes/ArtistSeries/ArtistSeriesMoreSeries.tsx:65]: wonder if it makes more sense to do the early return and not navigate here?

File: src/app/Scenes/Fair/FairArticles.tsx [hunk 3]
```diff
@@ -112,30 +115,31 @@ export const FairArticles: React.FC<FairArticlesProps> = ({ fair, relay }) => {
               return (
                 <Touchable
                   onPress={() => {
-                    navigate(article!.href!)
+                    if (!article?.href) return
+                    navigate(article?.href)
```
Comment (brainbicycle) [inline src/app/Scenes/Fair/FairArticles.tsx:119]: do we still need the ? in navigate with the early return?

**Review by brainbicycle — DISMISSED** [review]
some minor comments on unwrapping in navigate functions, otherwise looks great! thank you!

**Pull request merged**